platform/processor/live-instrument/src/test/kotlin/integration/CompositeDatatypesLiveBreakpointTest.kt

```diff
@@ -0,0 +1,169 @@
+        val instant = Instant.ofEpochSecond(1)
+        val localDate = LocalDate.now()
+        val localTime = LocalTime.of(1, 2, 3, 4)
+//        val offsetDateTime = OffsetDateTime.now()
+//        val offsetTime = OffsetTime.now()
+//        val zonedDateTime = ZonedDateTime.now()
+//        val zoneOffset = ZoneOffset.ofTotalSeconds(1)
+////        val zoneRegion = ZoneRegion("1",  ZoneRules.of(zoneOffset))
+        val bigInteger = BigInteger.TEN
+//        val clazz = Class.forName("Ezpeepee")
+
+        addLineLabel("done") { Throwable().stackTrace[0].lineNumber }
+        stopSpan(activeSpan)
+    }
+
+    @Test
+    fun `composite datatypes`() {
+        setupLineLabels { compositeDatatypes() }
+
+        val testContext = VertxTestContext()
+        val consumer = vertx.eventBus().consumer<Any>(toLiveInstrumentSubscriberAddress("system"))
+        consumer.handler {
+            val event = Json.decodeValue(it.body().toString(), LiveInstrumentEvent::class.java)
+            if (event.eventType == LiveInstrumentEventType.BREAKPOINT_HIT) {
+                //verify live breakpoint data
+                val breakpointHit =
+                    ProtocolMarshaller.deserializeLiveBreakpointHit(JsonObject(event.data))
+                testContext.verify {
+                    assertTrue(breakpointHit.stackTrace.elements.isNotEmpty())
+                    val topFrame = breakpointHit.stackTrace.elements.first()
+                    assertEquals(7, topFrame.variables.size)
+
+                    //todo: Check 1 test
+
+                    // Date
+                    assertEquals(
+                        Date(1L).toString(),
+                        topFrame.variables.find { it.name == "date" }!!.value.let { Date(it.toString()).toString() }
+                    )
+                    assertEquals(
+                        "java.util.Date",
+                        topFrame.variables.find { it.name == "date" }!!.liveClazz
+                    )
+
+                    // Duration
+                    assertEquals(
+                        Duration.ofSeconds(1).toString(),
+                        topFrame.variables.find { it.name == "duration" }!!.presentation
+                    )
+                    assertEquals(
+                        "java.time.Duration",
+                        topFrame.variables.find { it.name == "duration" }!!.liveClazz
+                    )
+
+                    // Instant
+                    assertEquals(
+                        Instant.ofEpochSecond(1).toString(),
+                        topFrame.variables.find { it.name == "instant" }!!.presentation
+                    )
+                    assertEquals(
+                        "java.time.Instant",
+                        topFrame.variables.find { it.name == "instant" }!!.liveClazz
+                    )
+
+                    // LocalDate
+                    assertEquals(
+                        LocalDate.now().toString(),
```

Comment:
`LocalDate.now()` will never equal the previous `LocalDate.now()`. You need to compare equivalent objects.